### PR TITLE
Add inspection for missing `Optional` default values

### PR DESCRIPTION
**Pull request by Lukellmann**
If a `dev.kord.common.entity.optional.Optional` is used in a `@Serializable` class, it should always have a default value so that deserialization works if the value is missing in the JSON.

Until now this was easy to forget which could lead to deserialization errors down the line.

By introducing a KSP processor that looks at all primary constructor parameters of `@Serializable` classes, there is now a static check executed at build time that verifies the default value is not forgotten in any place.

### CODE REVIEW

**Comment by Lukellmann** [comment]
ksp is nice, a compiler plugin with just 50 lines :)

File: ksp-processors/src/main/kotlin/inspection/OptionalDefaultInspectionProcessor.kt [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import kotlinx.serialization.Serializable
+
+/** [SymbolProcessorProvider] for [OptionalDefaultInspectionProcessor]. */
+class OptionalDefaultInspectionProcessorProvider : SymbolProcessorProvider {
+    override fun create(environment: SymbolProcessorEnvironment): SymbolProcessor =
+        OptionalDefaultInspectionProcessor(environment.logger)
+}
+
+private val OPTIONAL_TYPES =
+    listOf("Optional", "OptionalBoolean", "OptionalInt", "OptionalLong", "OptionalSnowflake")
```
[the remark this answers is not in the document]
Comment (Lukellmann) [reply ksp-processors/src/main/kotlin/inspection/OptionalDefaultInspectionProcessor.kt:17]: what don't you like about it?

[the remark this answers is not in the document]
Comment (DRSchlaubi) [reply ksp-processors/src/main/kotlin/inspection/OptionalDefaultInspectionProcessor.kt:17]: I just don't like having fully qualified names, I would prefer it to be type safe, by referencing the actual class

[the remark this answers is not in the document]
Comment (Lukellmann) [reply ksp-processors/src/main/kotlin/inspection/OptionalDefaultInspectionProcessor.kt:17]: hm, i don't think it's worth moving optionals to another module just for this

[the remark this answers is not in the document]
Comment (DRSchlaubi) [reply ksp-processors/src/main/kotlin/inspection/OptionalDefaultInspectionProcessor.kt:17]: I am kinda over-passionate about this, so you're probably right

**Pull request merged**